kafka_publisher/examples/twitter.py

```diff
@@ -72,11 +87,17 @@ def set_rules(rules: t.Any, auth: t.Any) -> None:
         )
 
 
-def stream_connect(auth: t.Any, kafka_producer: t.Any, topic: str) -> None:
+def stream_connect(
+    auth: t.Any, kafka_producer: t.Any, topic: str, token_bucket: TokenBucket
+) -> None:
     response = requests.get(STREAM_URL, auth=auth, stream=True)
     for response_line in response.iter_lines():
         if response_line:
-            kafka_producer.send(topic, response_line)
+            with TOKEN_BUCKET_LOCK:
+                if not token_bucket.num_tokens:
+                    continue
```

Comment:
wait this isn't quite right... my whole intent was to do clientside rate limiting; so what i need to do here is actually go to sleep and then let the `refill_token_bucket` daemon notify me when to come back awake

Reply:
This is a working prototype I think??
```
from dataclasses import dataclass
import threading 
import time

@dataclass
class TokenBucket:
    num_tokens: int


TOKEN_BUCKET_LOCK: threading.Condition = threading.Condition()

def refill_method(token_bucket: TokenBucket):
    while True:
        with TOKEN_BUCKET_LOCK:
            print("About to refill")
            time.sleep(15)
            token_bucket.num_tokens = 15
            print("Done refilling, notifying!")
            TOKEN_BUCKET_LOCK.notify()
        time.sleep(30)
        

def consume_method(token_bucket: TokenBucket):
    while True:
        with TOKEN_BUCKET_LOCK:
            if not token_bucket.num_tokens:
                print("Having to wait...")
                TOKEN_BUCKET_LOCK.wait()
            token_bucket.num_tokens -= 1
            print(f"Decrementing -- now is {token_bucket.num_tokens}")
            time.sleep(1)

token_bucket = TokenBucket(15)

t = threading.Thread(
    target=refill_method,
    args=(token_bucket,)
)
t.start()
consume_method(token_bucket)
```